Mathlib/Analysis/Meromorphic/NormalFormAt.lean

```diff
@@ -135,6 +124,143 @@ theorem AnalyticAt.meromorphicNFAt (hf : AnalyticAt 𝕜 f x) :
     MeromorphicNFAt f x := by
   simp [meromorphicNFAt_iff_analyticAt_or, hf]
 
+/-- Meromorphic functions have normal form outside of a discrete subset in the domain of
+meromorphicity. -/
+theorem MeromorphicOn.meromorphicNFAt_mem_codiscreteWithin [CompleteSpace E] {U : Set 𝕜}
+    (hf : MeromorphicOn f U) :
+    { x | MeromorphicNFAt f x } ∈ Filter.codiscreteWithin U := by
+  filter_upwards [hf.analyticAt_mem_codiscreteWithin] with _ ha
+  exact ha.meromorphicNFAt
+
+/-!
+## Vanishing and order
+-/
+
+/-- If `f` is meromorphic in normal form at `x`, then `f` has order zero iff it does not vanish at
+`x`.
+
+See `AnalyticAt.order_eq_zero_iff` for an analogous statement about analytic functions. -/
+theorem MeromorphicNFAt.order_eq_zero_iff (hf : MeromorphicNFAt f x) :
+    hf.meromorphicAt.order = 0 ↔ f x ≠ 0 := by
+  constructor
+  · intro h₁f
+    have h₂f := hf.order_nonneg_iff_analyticAt.1 (le_of_eq h₁f.symm)
+    apply h₂f.order_eq_zero_iff.1
+    apply ENat.map_natCast_eq_zero.1
+    rwa [h₂f.meromorphicAt_order] at h₁f
+  · intro h
+    have hf' := hf
+    rcases hf with h₁ | ⟨n, g, h₁g, h₂g, h₃g⟩
```

Comment:
Here is a trick to avoid needing to duplicate the hypothesis
```suggestion
    rcases id hf with h₁ | ⟨n, g, h₁g, h₂g, h₃g⟩
```

Reply:
Ah! There is so much notation to learn.